zephyr/include/rtos/cache.h

```diff
@@ -6,14 +6,63 @@
 #ifndef __ZEPHYR_RTOS_CACHE_H__
 #define __ZEPHYR_RTOS_CACHE_H__
 
-/* TODO: align with Zephyr generic cache API when ready */
 #define __SOF_LIB_CACHE_H__
-#include <arch/lib/cache.h>
+
+#if !defined(__ASSEMBLER__) && !defined(LINKER)
+
+#include <zephyr/cache.h>
+#include <zephyr/debug/sparse.h>
+
+#if defined(CONFIG_XTENSA) && defined(CONFIG_INTEL)
+
+/* definitions required by xtensa-based Intel platforms.
+ *
+ * TODO: if possible, move these to Zephyr.
+ */
+#define SRAM_UNCACHED_ALIAS 0x20000000
+#define is_cached(address) (!!((uintptr_t)(address) & SRAM_UNCACHED_ALIAS))
+
+#endif /* defined(CONFIG_XTENSA) && defined(CONFIG_INTEL) */
 
 /* writeback and invalidate data */
 #define CACHE_WRITEBACK_INV	0
 
 /* invalidate data */
 #define CACHE_INVALIDATE	1
 
+/* use SOF's wrapper over CONFIG_DCACHE_LINE_SIZE from Zephyr.
+ * This is because we don't want to force vendors to define
+ * CONFIG_DCACHE_LINE_SIZE inside their board configurations
+ * in order to avoid compilation errors.
+ *
+ * TODO: in time, switch to simply using CONFIG_DCACHE_LINE_SIZE
+ */
+#define DCACHE_LINE_SIZE CONFIG_SOF_DCACHE_LINE_SIZE
```

Comment:
I'm wondering how much more work would it be to go with `CONFIG_DCACHE_LINE_SIZE` directly?

Reply:
Hmm, shouldn't be much trouble. Will give it a shot and submit a Zephyr PR if I can get something working.

Reply:
@lyakh so should we merge this?